server/controller/authcontroller.js

```diff
@@ -1,74 +1,117 @@
-            const us=await user.save();
-            console.log(us)
-            res.status(200).json({"success":"Successfully registered"})
+            const social = await new Social(req.body.social)
+            social.save().then((social)=>{
+                socialId = social._id
+            })
+            if(parseInt(req.body.type)===1)
+            {
+                const user = new Organisation()
+                user.type = 1
+                user.name = req.body.name
+                user.adminName = req.body.adminName
+                user.website=(req.body.website?req.body.website:'')
+                user.pass = req.body.pass
+                user.followingList=[]
+                user.followersList=[]
+                user.email=req.body.email
+                user.location = req.body.location
+                user.social = socialId
+                const data=await user.save();
+                res.status(200).json({"success":"Successfully registered", status:1})
+            }
+            else {
+                const user=new User()
+                user.type = 0
+                user.name = req.body.name
+                user.dob=Date(req.body.dob);
+                user.gender=req.body.gender;
+                user.website=(req.body.website?req.body.website:'')
+                user.pass = req.body.pass
+                user.followingList=[]
+                user.followersList=[]
+                user.email=req.body.email
+                user.location = req.body.location
+                user.social = socialId
+                const data=await user.save();
+                res.status(200).json({
+                    "success":"Successfully registered",
+                    status:1
+                })
+            }
+            
         }
     },
 
     login:async(req,res)=>{
         const {error,isValid}=validateLoginInput(req.body);
-        if(!isValid)
-        {
-        return res.status(400).json(error);
-        }
-
-            const us=await User.findOne({email:req.body.email})
-              if(!us)
-              {
-              return  res.status(400).json({err:"USER NOT FOUND"})
-              }
-              else (us)
-              {
-              console.log(typeof(us))
-              console.log(us)
-              const k=JSON.parse(JSON.stringify(us))
-              let res2=await commonController.compare(req.body.password,k.pass)
-                if(res2==false)
-                {
-                    return res.status(400).json({"error":"Wrong password"})
-                }
-                else
-                {
-                    const payload={id:us._id};
-                    const tok=await jwt.sign(payload,secret,{expiresIn:3600})
-                      res.json({
-                        sucess:true,
-                        token:'Bearer '+tok
-                      })
-                }
-            } 
+        if(!isValid) return res.status(400).json({error, status:0});        
```

Comment:
search on best practices to use status codes.

Reply:
I have used 400 for bad request and 401 for auth error.